snapcraft/internal/db/datastore.py

```diff
@@ -0,0 +1,112 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2020 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import pathlib
+from typing import Any, Dict, List, Type
+
+import tinydb
+import yaml
+
+from . import errors, migration
+
+logger = logging.getLogger()
+
+
+class _YAMLStorage(tinydb.Storage):
+    """Provide YAML-backed storage for TinyDB."""
+
+    def __init__(self, path: str):
+        self.path = pathlib.Path(path)
+
+    def read(self) -> Dict[Any, Any]:
+        """Read database from file."""
+        try:
+            text_data = self.path.read_text()
+        except FileNotFoundError:
+            return dict()
+
+        return yaml.safe_load(text_data)
+
+    def write(self, data) -> None:
+        """Write database (data) to file."""
+        logger.debug(f"saving datstore: {self.path} {data}")
```

Comment:
why do we have a debug for writing but not one for reading?

Reply:
I think I just minimized output, fixed